apps/condo/domains/billing/schema/BillingAccount.test.js

```diff
@@ -510,4 +522,157 @@ describe('BillingAccount', () => {
+                const createdServiceConsumers = await ServiceConsumer.getAll(admin, {
+                    accountNumber: billingAccount.number,
+                    deletedAt: null,
+                })
+
+                expect(createdServiceConsumers).toHaveLength(2)
+                expect(createdServiceConsumers[0].organization.id).toEqual(organization.id)
+                expect(createdServiceConsumers[1].organization.id).toEqual(organization.id)
+                expect(createdServiceConsumers.map((serviceConsumer) => serviceConsumer.resident.id).toString()).toContain(resident.id)
+                expect(createdServiceConsumers.map((serviceConsumer) => serviceConsumer.resident.id).toString()).toContain(resident2.id)
+                expect(createdServiceConsumers[0].accountNumber).toEqual(billingAccount.number)
+                expect(createdServiceConsumers[1].accountNumber).toEqual(billingAccount.number)
+            })
+        })
+
+        test('should not create ServiceConsumer when BillingAccount is created if no resident exists for that property', async () => {
+            const unitName = faker.random.alphaNumeric(8)
+            const { address, addressMeta } = buildFakeAddressAndMeta(false)
+            const [organization] = await registerNewOrganization(admin)
+            const [billingProperty] = await createTestBillingProperty(admin, context, { address })
+            await createTestProperty(admin, organization, {
+                address,
+                addressMeta,
+            })
+
+            const [billingAccount] = await createTestBillingAccount(admin, context, billingProperty,
+                { unitName, unitType: FLAT_UNIT_TYPE }
+            )
+
+            await waitFor(async () => {
+                const createdServiceConsumers = await ServiceConsumer.getAll(admin, {
+                    accountNumber: billingAccount.number,
+                    deletedAt: null,
+                })
+
+                expect(createdServiceConsumers).toHaveLength(0)
+            })
+        })
+
+        test('should create ServiceConsumer only for passed BillingAccount', async () => {
+            const user1 = await makeClientWithResidentUser()
+            const unitName = faker.random.alphaNumeric(8)
+            const { address, addressMeta } = buildFakeAddressAndMeta(false)
+            const { address: address2 } = buildFakeAddressAndMeta(false)
+            const [organization] = await registerNewOrganization(admin)
+            const [acquiringIntegration] = await createTestAcquiringIntegration(admin)
+            await createTestAcquiringIntegrationContext(admin, organization, acquiringIntegration, { status: CONTEXT_FINISHED_STATUS })
+            const [billingProperty] = await createTestBillingProperty(admin, context, { address })
+            const [billingProperty2] = await createTestBillingProperty(admin, context, { address: address2 })
+            await createTestProperty(admin, organization, {
```

Comment:
Why do you need property, if it's not affecting DSC? Or not? Provide notes, more descriptive test name